pkg/kubelet/cm/container_manager_linux.go

```diff
@@ -240,6 +290,15 @@ func (cm *containerManagerImpl) setupNode() error {
 		return err
 	}
 
+	// Setup top level qos containers only if CgroupsPerQOS flag is specified as true
+	if cm.NodeConfig.CgroupsPerQOS {
+		qosContainersInfo, err := InitQOS(cm.NodeConfig.CgroupRoot, cm.subsystems)
```

Comment:
We talked about ensuring that qos cgroups continues to exist whenever a pod
level cgroup is being created. IIRC, @dubstack is going to add that
functionality in a future PR.

On Fri, Jul 8, 2016 at 2:38 PM, Derek Carr <notifications@github.com> wrote:

> In pkg/kubelet/cm/container_manager_linux.go
> <https://github.com/kubernetes/kubernetes/pull/27853#discussion_r70144806>
> :
>
> > @@ -240,6 +290,15 @@ func (cm *containerManagerImpl) setupNode() error {
> >  		return err
> >  	}
> >
> > +	// Setup top level qos containers only if CgroupsPerQOS flag is specified as true
> > +	if cm.NodeConfig.CgroupsPerQOS {
> > +		qosContainersInfo, err := InitQOS(cm.NodeConfig.CgroupRoot, cm.subsystems)
>
> for your own future sanity, i think you will want to ensure that these are
> never deleted by a rogue init system. its fine to handle that in a future
> todo if @vishh <https://github.com/vishh> agrees.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/kubernetes/kubernetes/pull/27853/files/8134b9de63922520d329b09348c13f5cc110507e#r70144806>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AGvIKM8pg3kWyFLjOpd4zq99F5RKmqdpks5qTsNBgaJpZM4I7o6F>
> .
>